### PR TITLE
LG-2126, LG-2650: replaces label with typography/Label

### PR DESCRIPTION
**Pull request by brookescarlett**
## ✍️ Proposed changes

🎟 _Jira ticket:_ [LG-2126](https://jira.mongodb.org/browse/LG-2126)
🎟 _Jira ticket:_ [LG-2650](https://jira.mongodb.org/browse/LG-2650)

## 🛠 Types of changes

- Replaces label with typography/Label
- Refactors to new directory structure

- [ ] Tooling (updates to workspace config or internal tooling)
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Style change (purely visual updates; if this is the case, attach a screenshot below!)

## ✅ Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have run `yarn changeset` and documented my changes

- [ ] I have added my new package to the global tsconfig
- [ ] I have added my new package to the Table of Contents on the global README

## Feedback Levels

Please try to mark your comments with one of these levels

1. Praise (👏 🚀 🔥 🍾)
2. Opinions, Ideas, Suggestions (💡 👀)
3. Let’s discuss @ Code review (🤷 🤔 🧐 )
4. Doesn’t meet agreed upon standards (😬 ⚠️ ‼️)
5. Missing an edge case (🛑 🙅 ❌)
6. Missing core functionality (⛔️ ☣️ 🚧 🌋 💀)

(Note: Use ⌘-⌃-␣ to open the Emoji keyboard!)

### CODE REVIEW

File: packages/radio-group/src/Radio/Radio.types.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  default?: boolean;
+
+  /**
+   * Boolean that determines if the Radio is disabled.
+   */
+  disabled?: boolean;
+  /**
+   * Content that will appear inside of Radio.
+   */
+  children?: React.ReactNode;
```
Comment (spark33) [inline packages/radio-group/src/Radio/Radio.types.ts:24]: nit: we could use `PropsWithChildren`

**Review by spark33 — APPROVED** [review]

File: packages/radio-group/src/Radio/styles.ts [hunk 2]
```diff
@@ -22,37 +20,11 @@ export const containerMargin = css`
...
-};
-
 export const labelBaseStyle = css`
   display: flex;
   line-height: 20px;
   cursor: pointer;
   align-items: flex-start;
-  font-size: ${typeScales.body1.fontSize}px;
   font-weight: 700;
 `;
```
Comment (spark33) [inline packages/radio-group/src/Radio/styles.ts:30]: This file should be named `Radio.styles.ts`

**Review by spark33 — CHANGES_REQUESTED** [review]

File: packages/radio-group/src/Radio/Radio.types.ts (hunk 1 above)
Comment (brookescarlett) [reply packages/radio-group/src/Radio/Radio.types.ts:24]: Does that still exist with React 18?

File: packages/radio-group/src/Radio/styles.ts (hunk 2 above)
Comment (brookescarlett) [reply packages/radio-group/src/Radio/styles.ts:30]: Good catch

**Review by spark33 — APPROVED** [review]

**Pull request merged**